buildSrc/src/main/kotlin/cleanup.gradle.kts

```diff
@@ -0,0 +1,105 @@
+check(rootProject.name == name) {
+    "The cleanup plugin should be applied to the root project and not $name"
+}
+
+tasks.register("templateCleanup") {
+    val repository = System.getenv("GITHUB_REPOSITORY")
+        ?: error("No GITHUB_REPOSITORY environment variable. Are you running from Github Actions?")
+
+    val (owner, name) = repository.split("/").let {
+        it[0].sanitized() to it[1].sanitized()
+    }
+
+    file("settings.gradle.kts").replace(
+        "rootProject.name = (\"kotlin-android-template\")",
+        "rootProject.name = (\"$name\")"
+    )
+    file("buildSrc/src/main/java/Coordinates.kt").replace(
+        "com.ncorti.kotlin.template",
+        "com.github.$owner.$name"
+    )
+    /**
+     * This fails with: 'refusing to allow a GitHub App to create or update workflow
+     * `.github/workflows/publish-release.yaml` without `workflows` permission'
+     */
+    /*
+    file(".github/workflows/publish-release.yaml").replace(
+        "cortinico/kotlin-android-template",
+        "$owner/$name"
+    )
+    file(".github/workflows/publish-snapshot.yaml").replace(
+        "cortinico/kotlin-android-template",
+        "$owner/$name"
+    )
+     */
+    file("buildSrc/src/main/kotlin/publish.gradle.kts").apply {
+        replace("cortinico/kotlin-android-template", "$owner/$name")
+        replace("cortinico/kotlin-android-template", "$owner/$name")
+        replace("cortinico", owner)
+        replace("Nicola Corti", owner)
+        // Keep the link to the original script
+        replace(
+            "* https://github.com/$owner/$name/blob/master/buildSrc/src/main/kotlin/publish.gradle.kts",
+        "* https://github.com/cortinico/kotlin-android-template/blob/master/buildSrc/src/main/kotlin/publish.gradle.kts"
+        )
+    }
+
+    copyTemplateFiles(repository, name)
```

Comment:
I'm unsure that we want to wipe the old README. It contains somehow useful information for the setup 🤔 So I'd rather `cat` the old readme after the one you added, what do you think?

Reply:
> The reason why I initially disabled it was to make sure that once you start using the template the CI is green and you don't have to understand why it fails.

Makes sense. So maybe we leave these files untouched and add a comment?

```
jobs:
  publish:
    # remove this check when your secrets are setup and you're ready to publish
    if: ${{ github.repository == 'cortinico/kotlin-android-template'}}
```

> I'm unsure that we want to wipe the old README. It contains somehow useful information for the setup 🤔 So I'd rather cat the old readme after the one you added, what do you think?

Yea, agreed the existing README is way more appealing. What about replacing the introduction of the template README  (everything that's before "Features") with the new README and checklist?

Reply:
Yup makes sense. We can adapt the "How to use" section.

Reply:
I made the merge, this is how it looks: https://github.com/logingithubtest/test6